### PR TITLE
(feat): new route for querying tags

### PR DESCRIPTION
**Pull request by Thiemann96**
Added simple Route to retrieve all unique tags that are not empty. Used to give suggestions for tags. Results will be cached for 5 minutes.

## Description
New `GET /getAllTags` route which gets all boxes, filters out tags, and response with an array of tags. 
 
## Motivation and Context

Used for suggesting new tags to users. Will be implemented in openSenseMapX as autocomplete at registration. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code has been linted using `yarn run lint`.
- [X] My code does not break the tests (`yarn run test`)
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: packages/api/lib/controllers/boxesController.js [hunk 1]
```diff
@@ -487,6 +487,26 @@ const deleteBox = async function deleteBox (req, res, next) {
   }
 };
 
+const getAllTags = async function getAllTags (req, res, next) {
+  try {
+    const output = [];
+    const boxes = await Box.find().exec();
```
Comment (Thiemann96) [inline packages/api/lib/controllers/boxesController.js:493]: Pretty sure there is a "cleaner" way of doing this , by only returning grouptags in the first place, but for now I have not been able to find this. Will use this for testing purposes on OSMX development, but I am happy for any input concering this

Comment (mpfeil) [reply packages/api/lib/controllers/boxesController.js:493]: You can use the `distinct` functionality of the `boxes` collection.
```javascript
const grouptags = await Box.find().distinct('grouptag').exec();
or
const grouptags2 = await Box.collection.distinct('grouptag');
```

Comment (mpfeil) [reply packages/api/lib/controllers/boxesController.js:493]: You can use the `distinct` functionality of the `boxes` collection.
```javascript
const grouptags = await Box.find().distinct('grouptag').exec();
or
const grouptags2 = await Box.collection.distinct('grouptag');
```

File: packages/api/lib/routes.js [hunk 2]
```diff
@@ -74,6 +74,7 @@ const routes = {
   'noauth': [
     { path: '/', method: 'get', handler: printRoutes, reference: 'api-Misc-printRoutes' },
     { path: '/stats', method: 'get', handler: statisticsController.getStatistics, reference: 'api-Misc-getStatistics' },
+    { path: '/getAllTags', method: 'get', handler: boxesController.getAllTags, reference: 'api-Boxes-getAllTags' },
```
Comment (mpfeil) [inline packages/api/lib/routes.js:77]: Just call the route `/tags`.
The method already indicates that it is a `GET` request.
`All` is specifying a parameter. If it is needed pass it as a parameter like `?all=true`

**Review by mpfeil — CHANGES_REQUESTED** [review]

**Comment by mpfeil** [comment]
I have converted the PR to a draft pull request. If you are finished just remove the `work in progress` label and convert it back by clicking `Ready for review`.

**Review by mpfeil — APPROVED** [review]

**Pull request merged**